lib/IRGen/GenClass.cpp

```diff
@@ -243,6 +243,25 @@ namespace {
           // the field offset vector.
           ClassHasFixedSize = false;
 
+          // We can't use global offset variables if we are generic and layout
+          // dependent on a generic parameter because the objective-c layout might
+          // depend on the alignment of the generic stored property('t' in the
+          // example below).
+          //
+          // class Foo<T> : NSFoobar {
+          //   var x : AKlass = AKlass()
+          //   var y : AKlass = AKlass()
+          //   var t : T?
+          // }
+          if (classType.hasArchetype())
+            for (VarDecl *var : theClass->getStoredProperties()) {
+              SILType type = classType.getFieldType(var, IGM.getSILModule());
+              auto &eltType = IGM.getTypeInfo(type);
+              if (!eltType.isFixedSize()) {
+                if (type.hasArchetype())
```

Comment:
The preceding !eltType.isFixedSize() check excludes the latter.

In the resilient case it is okay to use global offset variables because there exists only one type (Type Foo with resilient var x: R) not many like in the generic case (Type family Foo<T> with var x: T) that get mapped to the global offset.